ktlint-cli/build.gradle.kts

```diff
@@ -51,37 +51,36 @@ val shadowJarExecutable by tasks.registering(DefaultTask::class) {
 
     dependsOn(tasks.shadowJar)
 
-    val isReleaseBuild = !version.toString().endsWith("SNAPSHOT")
```

Comment:
❗ 
I couldn't find a reason why snapshot would not need to have the executable files written 👀 Please let me know if I'm missing something. 
With the new config:
- executable files will always be created
- signature files will be created only if signing is enabled (driven by [this](https://github.com/pinterest/ktlint/blob/013b8a66506d1d278205da72e4ee925fdcb67999/build-logic/src/main/kotlin/ktlint-publication.gradle.kts#L105) logic)